go/vt/topo/consultopo/lock.go

```diff
@@ -20,20 +20,41 @@ import (
 	"fmt"
 	"path"
 
-	log "github.com/golang/glog"
 	"github.com/hashicorp/consul/api"
 	"github.com/youtube/vitess/go/vt/topo"
 	"golang.org/x/net/context"
 )
 
-func (s *Server) lock(ctx context.Context, c *cellClient, lockPath, contents string) (string, error) {
+// consulLockDescriptor implements topo.LockDescriptor.
+type consulLockDescriptor struct {
+	s        *Server
+	c        *cellClient
+	lockPath string
+}
+
+// Lock is part of the topo.Backend interface.
+func (s *Server) Lock(ctx context.Context, cell string, dirPath string) (topo.LockDescriptor, error) {
+	// We list the directory first to make sure it exists.
+	if _, err := s.ListDir(ctx, cell, dirPath); err != nil {
+		if err == topo.ErrNoNode {
+			return nil, err
+		}
+		return nil, fmt.Errorf("cannot ListDir(%v,%v) before locking", cell, dirPath)
+	}
+
+	c, err := s.clientForCell(ctx, cell)
+	if err != nil {
+		return nil, err
+	}
+	lockPath := path.Join(c.root, dirPath, locksFilename)
+
 	// Build the lock structure.
-	l, err := s.global.client.LockOpts(&api.LockOptions{
+	l, err := c.client.LockOpts(&api.LockOptions{
 		Key:   lockPath,
-		Value: []byte(contents),
+		Value: []byte("lock"),
```

Comment:
Why did you abandon the idea of writing contents into the lock?

Reply:
I re-added it. I thought it would be simpler, but the contents is useful in a lot of topo implementations.